File: split/runaway.pas
procedure runaway;
var
  p: halfword;
begin
  if scannerstatus > 1 then
  begin
    printnl(569);
    case scannerstatus of
      2:
      begin
        print(570);
        p := defref;
      end;
      3:
      begin
        print(571);
        p := 29997;
      end;
      4:
      begin
        print(572);
        p := 29996;
      end;
      5:
      begin
        print(573);
        p := defref;
      end;
    end;
    printchar(63);
    println;
    showtokenlist(mem[p].hh.rh, 0, errorline - 10);
  end;
end;

File: cpp/runaway.cpp
#include "runaway.h"
#include "showtokenlist.h"
#include "print.h"
#include "printnl.h"
#include "printchar.h"
#include "println.h"

void runaway(void)
{
	if (scannerstatus > skipping)
	{
		printnl("Runaway ");
		halfword p;
		switch (scannerstatus)
		{
			case defining:
				print("definition");
				p = defref;
				break;
			case matching:
				print("argument");
				p = temp_head;
				break;
			case aligning:
				print("preamble");
				p = hold_head;
				break;
			case absorbing:
				print("text");
				p = defref;
		}
		printchar('?');
		println();
		showtokenlist(link(p), 0, errorline-10);
	}
}
